core/services/ocr2/plugins/ocr2keeper/evm21/logprovider/buffer.go

```diff
@@ -243,49 +230,42 @@ func (b *logEventBuffer) dequeueRange(start, end int64, upkeepLimit int) []fetch
 // NOTE: this function should be called with the lock held
 func (b *logEventBuffer) getBlocksInRange(start, end int) []fetchedBlock {
 	var blocksInRange []fetchedBlock
-	start, end = b.normalRange(start, end)
+	start, end = b.blockRangeToIndices(start, end)
 	if start == -1 || end == -1 {
 		// invalid range
 		return blocksInRange
 	}
-	if start < end {
-		return b.blocks[start:end]
+	if start <= end {
+		// Normal range, need to return indices from start to end(inclusive)
+		return b.blocks[start : end+1]
```

Comment:
verified in docs https://go.dev/ref/spec#Slice_expressions that it is safe to do `b.blocks[start : end+1]` when `end = len(b.blocks)-1` (last index)